src/minifig_detector.py

```diff
@@ -0,0 +1,196 @@
+        class_ids = []
+
+        # 各検出候補を処理
+        for i in range(num_boxes):
+            max_score = -1.0
+            best_class = -1
+
+            # クラススコアの最大値とクラスIDを取得
+            for j in range(num_classes):
+                score = data[4 + j, i]
+                if score > max_score:
+                    max_score = score
+                    best_class = j
+
+            # 信頼度閾値による候補フィルタリング
+            if max_score < self.conf_threshold:
+                continue
+
+            # バウンディングボックス座標抽出（YOLO形式：中心座標＋幅高さ）
+            cx = data[0, i]  # 中心X座標（640x640空間）
+            cy = data[1, i]  # 中心Y座標（640x640空間）
+            w = data[2, i]   # 幅（640x640空間）
+            h = data[3, i]   # 高さ（640x640空間）
+
+            # レターボックス座標系から元画像座標系に逆変換
+            center_x = int((cx - pad[0]) / scale)  # パディング補正＋スケール逆変換
+            center_y = int((cy - pad[1]) / scale)
+            width = int(w / scale)
+            height = int(h / scale)
+
+            # 中心座標形式から左上座標形式（OpenCV形式）に変換
+            left = center_x - width // 2
+            top = center_y - height // 2
+
+            # NMS用データに追加
+            boxes.append([left, top, width, height])  # [x,y,w,h]形式
+            confidences.append(max_score)  # 最高クラススコア
+            class_ids.append(best_class)  # 最高スコアのクラスID
+
+        # 信頼度閾値を満たす検出がない場合
+        if not boxes:
+            return {"wasDetected": False,
+                    "direction": "NONE", "confidence": 0.0}
+
+        # Non-Maximum Suppression で重複検出を除去
+        indices = cv2.dnn.NMSBoxes(
+            boxes, confidences, self.conf_threshold, self.nms_threshold)
+
+        # NMS後に有効な検出が残っている場合
+        if len(indices) > 0:
+            best_idx = indices.flatten()[0]  # 最も信頼度の高い検出を選択
+            best_class_id = class_ids[best_idx]  # 対応するクラスID
+            best_confidence = confidences[best_idx]  # 対応する信頼度
+            direction = self.labels[best_class_id]  # ミニフィグの向き
+
+            return {
+                "wasDetected": True,
+                "direction": direction,  # "front", "back", "right", "left"
+                "confidence": float(best_confidence)  # 0.0-1.0の信頼度
+            }
+
+        return {"wasDetected": False, "direction": "NONE", "confidence": 0.0}
+
+    def detect(self, image_path: str) -> dict:
+        """minifigの向きを判定.
+
+        Args:
+            image_path (str): 画像ファイルのパス
+
+        Returns:
+            dict: 検出結果 {"wasDetected": bool, "direction": str,
+                         "confidence": float}
+        """
+        # 共通エラーレスポンス
+        error_result = {"wasDetected": False,
+                        "direction": "NONE", "confidence": 0.0}
+
+        # 入力ファイル存在チェック
+        if not os.path.exists(image_path):
+            return error_result
+
+        # モデル読み込み状態チェック
+        if self.session is None:
+            return error_result
+
+        # 画像読み込み（BGR形式）
+        img = cv2.imread(image_path)
+        if img is None:  # 読み込み失敗（不正ファイル等）
+            return error_result
```

Comment:
エラーレスポンスは共通でいいと思うけど、何のエラーでやらかしたのか知りたいから何かしら出力してくれると助かる

```suggestion
        # 入力ファイル存在チェック
        if not os.path.exists(image_path):
            print("入力ファイルが存在しません")
            return error_result

        # モデル読み込み状態チェック
        if self.session is None:
            print("モデル読み込みエラー")        
            return error_result

        # 画像読み込み（BGR形式）
        img = cv2.imread(image_path)
        if img is None:  # 読み込み失敗（不正ファイル等）
            print("画像読み込みエラー")
            return error_result
```